### PR TITLE
Expose 2 symbols from android_main

### PR DESCRIPTION
**Pull request by panos-lunarg**
MainGetCurrentBlockIndex and MainGetLoadingTrimmedState whould de discoveralbe through dlopen + dlsym from the android replay native .so:

- MainGetCurrentBlockIndex(): Returns the current block index as calculated from the file processor

- MainGetLoadingTrimmedState(): Returns wheter file processor is between a FrameStateBegin/FrameStateEnd markers pair

### CODE REVIEW

File: framework/decode/vulkan_replay_consumer_base.cpp [hunk 1]
```diff
@@ -7440,13 +7493,11 @@ VkResult VulkanReplayConsumerBase::OverrideCreateAccelerationStructureKHR(
                 capture_id);
         }
 
-        result = device_table->CreateAccelerationStructureKHR(
-            device, &modified_create_info, GetAllocationCallbacks(pAllocator), replay_accel_struct);
+        result = func(device, &modified_create_info, GetAllocationCallbacks(pAllocator), replay_accel_struct);
```
Comment (bradgrantham-lunarg) [inline framework/decode/vulkan_replay_consumer_base.cpp:7496]: This doesn't seem related to the PR purpose - is this and the other changes to the RT calls where you pass in `func` a bugfix unrelated to the block number and marking generated commands?

Comment (panos-lunarg) [reply framework/decode/vulkan_replay_consumer_base.cpp:7496]: I was looking for injected commands by searching for places where `device_table` is referenced to fetch the api call function pointer. While doing so I discovered these cases where an api function pointer is fetched from `device_table` unnecessarily while the pointer is already available.
 
Strictly speaking this is not 100% related to PR's purposes but, given the opportunity, I clean up those places. I can revert if you insist.

File: framework/util/mark_injected_commands.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    BeginInjectCommands_fp = begin_fp;
+    EndInjectCommands_fp   = end_fp;
+}
+
+void BeginInjectedCommands()
+{
+#ifdef GFXRECON_DEBUG_BUILD
+    if (injecting_api_calls_g)
+    {
+        GFXRECON_LOG_WARNING("Nested BeginInjectedCommands/EndInjectedCommands");
```
Comment (davidd-lunarg) [inline framework/util/mark_injected_commands.cpp:51]: Nice error check. My understanding of it is that if we hit this message it means that GFXR code is incorrectly using the Begin/End calls. If that is the case this could be upgraded to a `GFXRECON_ASSERT` since it indicates a code problem, not a runtime problem.

File: framework/util/mark_injected_commands.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+GFXRECON_BEGIN_NAMESPACE(util)
+
+static void BeginEndInjectedCommandsNoop() {}
+
+#ifdef GFXRECON_DEBUG_BUILD
+static bool injecting_api_calls_g = false;
+#endif
+
+static PFN_BeginInjectedCommands BeginInjectCommands_fp = BeginEndInjectedCommandsNoop;
+static PFN_EndInjectedCommands   EndInjectCommands_fp   = BeginEndInjectedCommandsNoop;
```
Comment (davidd-lunarg) [inline framework/util/mark_injected_commands.cpp:38]: Will modules outside of `decode` need this utility class? If not then moving them to the `decode` namespace and directory could help with code organization, and I think it would prevent the static variables and `extern "C"` method from existing in non-decode libraries that use the `util` library.

**Review by davidd-lunarg — APPROVED** [review]

File: framework/util/mark_injected_commands.cpp (hunk 3 above)
Comment (panos-lunarg) [reply framework/util/mark_injected_commands.cpp:38]: good point

File: framework/decode/mark_injected_commands.cpp
Comment (dfriederich) [inline framework/decode/mark_injected_commands.cpp:1]: I wonder if we should have some header/place where all exported api's are declared with some docu.
Right now the exported functions are defined in different cpp files, and if I don't know what to look out for I would not find any connection. (well maybe there is no connection :-) ).

**Comment by panos-lunarg** [comment]
@dfriederich All your comments have good points. I hoped I addressed all of them

File: framework/decode/vulkan_replay_consumer_base.cpp (hunk 1 above)
Comment (bradgrantham-lunarg) [reply framework/decode/vulkan_replay_consumer_base.cpp:7462]: No, sounds good to me.  In the future it is slightly helpful to separate clean ups into separate PRs so I can go through those more quickly.

**Review by bradgrantham-lunarg — APPROVED** [review]
